docs/dev/adding_instruments.rst

```diff
@@ -732,6 +644,102 @@ In the above example, :code:`MultimeterA` and :code:`MultimeterB` use a differen
+
+The getter of :meth:`Instrument.control <pymeasure.instruments.Instrument.control>` does not call them directly, but via a chain of methods. It calls :meth:`~pymeasure.instruments.Instrument.values` which in turn calls :meth:`~pymeasure.instruments.Instrument.ask` and processes the returned string into understandable values. :meth:`~pymeasure.instruments.Instrument.ask` sends the readout command via :meth:`write`, waits some time if necessary via :meth:`wait_till_read`, and reads the device response via :meth:`read`.
+
+Similarly, :meth:`Instrument.binary_values <pymeasure.instruments.Instrument.binary_values>` sends a command via :meth:`write`, waits with :meth:`wait_till_read`, but reads the response via :meth:`Adapter.read_binary_values <pymeasure.adapters.Adapter.read_binary_values>`.
+
+
+Adding a device address
+***********************
+
+Let's look at a simple example for a device, which requires its address as the first three characters and returns the same style. This is straightforward, as :meth:`write` just prepends the device address to the command, and :meth:`read` has to strip it again doing some error checking. Similarly, a checksum could be added.
+
+.. code-block:: python
+
+    class Extreme5002(Instrument):
+        """The great Extreme5002 instrument.
+        :param address: The device address for the communication.
+        """
+        def __init__(self, adapter, address=0):
+            super().__init__(adapter, "Extreme5002")
+            self.address = f"{address:03}"
+
+        def write(self, command):
+            """Add the device address in front of every command and send it to the device."""
+            super().write(self.address + command)
+
+        def read(self):
+            """Read from the device and assert, that the response starts with the device address."""
+            got = super().read()
+            if got.startswith(self.address):
+                return got[3:]
+            else:
+                raise ConnectionError(f"Message {got} for wrong address read.")
+    
+        voltage = Instrument.measurement(
+            ":VOLT:?", "The measured voltage in Volts.")
+
+If the device is initialized with :code:`address=12`, a request for the voltage would send :code:`"012:VOLT:?"` to the device and expect a response beginning with :code:`"012"`.
+
+
+Bytes communication
+*******************
+
+Some devices do not expect ASCII strings but raw bytes. In those cases, you can call the :meth:`write_bytes` and :meth:`read_bytes` in your :meth:`write` and :meth:`read` methods. The following example shows an instrument, which has registers to be written and read via bytes sent.
+
+.. code-block:: python
+
+    class Extreme5003(Instrument):
```

Comment:
I love this example :star_struck: ! I think we can now lay #296 to rest!
Maybe call the instrument `ExtremeBytes` :D ? There is no obligation to carry on a number-based theme...

Reply:
Thanks.
I tested the idea with the HCP TC38D (PR is going to follow later) and took the resulting code as this example.